### PR TITLE
Added Theorem 8.30

### PR DESCRIPTION
**Pull request by nimarasekh**
Partially addresses #18.

### CODE REVIEW

File: src/simplicial-hott/08-covariant.rzk.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ This is a literate `rzk` file:
   instance the notion of contractible types.
 - `3-simplicial-type-theory.md` — We rely on definitions of simplicies and their
   subshapes.
+- `4-extension-types.md` — We use Theorem 4.1, an equivalence between lifts.
```
Comment (fizruk) [inline src/simplicial-hott/08-covariant.rzk.md:17]: I assume, you would also want `WeakExt` from there after #42 is merged.

Comment (nimarasekh) [reply src/simplicial-hott/08-covariant.rzk.md:17]: I agree the WeakExt I wrote was temporary and meant to replaced by the appropriate one in the right place. I just didn't where that would be. So, yeah we should certainly adjust it.

File: src/simplicial-hott/08-covariant.rzk.md [hunk 2]
```diff
@@ -1682,3 +1683,31 @@ commuting with the contravariant lifts.
...
+        ( \ x y f g →  
+           equiv-with-contractible-codomain-implies-contractible-domain
+           ( (t : Δ¹) → ((b : B) → C (f t) b) [  t ≡ 0₂ ↦ g ])   
+           ( (b : B) → (t : Δ¹) → C (f t) b [ t ≡ 0₂ ↦ g b]) 
+           ( flip-ext-fun 2 Δ¹ (\ t → t ≡ 0₂) B ( \ t →  C (f t)) ( \ t → g)) 
+          (we 
+            ( \ b → ( (t : Δ¹) → C (f t) b [ t ≡ 0₂ ↦ g b] ) ) 
+            ( \ b → ( (( second ( has-unique-fixed-domain-lifts-iff-is-covariant A ( \ a  → (C a b) )) ) (is-locally-covariant b)) x y f (g b) ) )
+            )
+        )
```
Comment (fizruk) [inline src/simplicial-hott/08-covariant.rzk.md:1711]: Here, I assume it'd be easier (and more helpful) to talk personally to me or @emilyriehl about the [Code style](https://rzk-lang.github.io/sHoTT/STYLEGUIDE/). But if you want you can try adjusting this yourself before we can meet :)

Comment (nimarasekh) [reply src/simplicial-hott/08-covariant.rzk.md:1711]: Thanks for the link. I'll give it a shot based on the link and then make sure to get feedback afterwards when we meet in the coming days.

**Comment by emilyriehl** [comment]
I just merged the weak-ext-ext stuff into main and into this branch. @nimarasekh now you have WeakExtExt at your disposal.

**Comment by nimarasekh** [comment]
Added weak function extentensionality in separate file (06-contractible) that is now "assumed" in the covariant family file.  So now it should read cleaner.

**Comment by emilyriehl** [comment]
Hey @nimarasekh. I just made some stylistic edits to the formatting (fixing indentations, tweaking comments). Then I also broke up the logical equivalence between the two notions of is-covariant into separate functions that can be applied with out calling `first iff` and `second iff`.

**Review by emilyriehl — APPROVED** [review]
Good to go.

**Pull request merged**